src/pocketmine/network/mcpe/PlayerNetworkSessionAdapter.php

```diff
@@ -239,7 +239,37 @@ public function handleBookEdit(BookEditPacket $packet) : bool{
 	}
 
 	public function handleModalFormResponse(ModalFormResponsePacket $packet) : bool{
-		return $this->player->onFormSubmit($packet->formId, json_decode($packet->formData, true));
+		return $this->player->onFormSubmit($packet->formId, self::stupid_json_decode($packet->formData, true));
+	}
+
+	/**
+	 * Hack to work around a stupid bug in Minecraft W10 which causes empty strings to be sent unquoted in form responses.
+	 *
+	 * @param string $json
+	 * @param bool   $assoc
+	 *
+	 * @return mixed
+	 */
+	private static function stupid_json_decode(string $json, bool $assoc = false){
```

Comment:
Why not to make it public? Let other plugins fix it too

Reply:
There's no need for other plugins to fix it.

Reply:
Why? Many have not yet switched from custom forms api to pm api

Reply:
This is a solution for the new API of 3.2. Plugins should be upgrading to use the new API anyway.

Reply:
By "new API" I'm referring to the `Form` interface and `Player->sendForm()`. Here's an example changeset: https://github.com/jojoe77777/FormAPI/pull/41

Reply:
How about third party APIs?

Reply:
> Plugins should be upgrading to use the new API anyway.